package/base-files/files/lib/upgrade/common.sh

```diff
@@ -277,7 +277,7 @@ get_partitions() { # <device> <filename>
 		local magic=$(dd if="$disk" bs=2 count=1 skip=255 2>/dev/null)
 		if [ "$magic" != $'\x55\xAA' ]; then
 			v "Invalid partition table on $disk"
-			exit
+			return 1
```

Comment:
none of the callers is checking the return value of this function. Will we still abort in case of an error?